imap/reconstruct.c

```diff
@@ -454,6 +459,19 @@ static int do_reconstruct(struct findall_data *data, void *rock)
 
     name = mbname_intname(data->mbname);
 
+    if (reconstruct_flags & RECONSTRUCT_REPAIR_MBOXLIST) {
+        /* Skip tombstones */
+        if (!(data->mbentry->mbtype & MBTYPE_DELETED)) {
+            mbentry_t *mbentry = NULL;
+            r = mboxlist_lookup_by_uniqueid(data->mbentry->uniqueid, &mbentry, NULL);
+            if (r == IMAP_MAILBOX_NONEXISTENT) {
+                mboxlist_update((struct mboxlist_entry *) data->mbentry, 1);
+            }
```

Comment:
We should also check that the entry matches, and repair if it doesn't.  Same "fix up the things" argument.